package/ports_daemon/ports_daemon/src/ports.c

```diff
@@ -97,17 +97,21 @@ typedef struct port_config_s {
   pid_t adapter_pid; /* May be cleared by SIGCHLD handler */
   restart_type_t restart;
   bool first_start;
+  const int blacklist_len;
+  const char *const blacklist[10];
+  int mode_mapping[10];           /* maps from global mode list to port mode sublist */
```

Comment:
Would it be simpler to just use the global mode mapping list and just mark certain indexes as invalid?  This would allow us to not have to maintain this sublist mapping indexes->mode names.

Reply:
oh, thats a good idea.